### PR TITLE
Enable checkstyle for google style guide

### PR DESCRIPTION
**Pull request by trask**
Most of #1514

I'll send another PR afterwards to enable `OverloadMethodsDeclarationOrder` as it requires moving a few methods around which makes for hard to review diffs.

EDIT: also will send separate PR to enable/discuss `ClassTypeParameterName` and `MethodTypeParameterName`

### CODE REVIEW

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/DatabaseClientTracer.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
 import java.net.InetSocketAddress;
 import java.util.concurrent.ExecutionException;
 
-public abstract class DatabaseClientTracer<CONNECTION, QUERY> extends BaseTracer {
+public abstract class DatabaseClientTracer<ConnectionT, QueryT> extends BaseTracer {
```
Comment (iNikem) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/DatabaseClientTracer.java:21]: Just a random pick: not sure I like this change :(

File: instrumentation/logback/logback-1.0/javaagent/logback-1.0.0-javaagent.gradle [hunk 2]
```diff
@@ -9,11 +9,11 @@ muzzle {
 }
 
 dependencies {
-  implementation project(':instrumentation:logback:logback-1.0.0:library')
+  implementation project(':instrumentation:logback:logback-1.0:library')
```
Comment (mateuszrzeszutek) [inline instrumentation/logback/logback-1.0/javaagent/logback-1.0.0-javaagent.gradle:12]: Hm, shouldn't you rename this file to `logback-1.0-javaagent.gradle` as well?

**Review by mateuszrzeszutek — APPROVED** [review]
Nice! 🎉  I love CamelCase consistency in all class/method names.

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/DatabaseClientTracer.java (hunk 1 above)
Comment (trask) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/DatabaseClientTracer.java]: I reverted the type variable names, and will submit a separate PR to discuss those changes (I had done that change as a separate commit suspecting potential objection 😁)

**Comment by trask** [comment]
Going to merge before conflicts start building up. Not a problem to revisit any of the checkstyle rules if you find you don't like something later on.

**Pull request merged**